tests/unit/validation/test_install_validation.py

```diff
@@ -403,6 +403,26 @@ def test_pubsub_config(mocker):
     }
 
 
+def test_admins(mocker):
+    assert validate_install_configuration(
```

Comment:
Thanks for adding the test here.
Unfortunately it doesn't actually "assert" that the config is being parsed with no errors.

This happens because the install YAML will be used even if there are errors in it (see [here](https://github.com/codecov/shared/blob/70039b5b9e5cc1466041da351a07ebf9546fe89a/shared/validation/install.py#L363))

We use this assert to verify that nothing was modified by the validator unexpectedly, but to test that it was validated with no error we actually mock the logger and check if no warnings were emitted. Like [here](https://github.com/codecov/shared/blob/70039b5b9e5cc1466041da351a07ebf9546fe89a/tests/unit/validation/test_install_validation.py#L336)

All this to say, could you please change the test to
 
```python
mock_warning = mocker.patch.object(install_log, "warning")
# validate_install_configuration being called + assert of output
assert mock_warning.call_count == 0
```

Reply:
Oh, gotcha. That makes sense. I just adopted the other testing pattern for this PR.

Reply:
Hum... I see. That other test should not be like that either. I'll make note to fix that later.